ibm/service/iampolicy/data_source_ibm_iam_policy_assignments.go

```diff
@@ -0,0 +1,308 @@
+// Copyright IBM Corp. 2023 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package iampolicy
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM/platform-services-go-sdk/iampolicymanagementv1"
+)
+
+func DataSourceIBMIAMPolicyAssignments() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIBMIAMPolicyAssignmentsRead,
+
+		Schema: map[string]*schema.Schema{
+			"account_id": {
```

Comment:
To be in sync with all other res/ds can we remove accoutn_id as a input parameter and fetch account_id from logged in user context
```
	userDetails, err := meta.(conns.ClientSession).BluemixUserDetails()
	if err != nil {
		return err
	}
	userDetails.UserAccount
	```